Add `aws iam group policy attachment`

eliecharra opened the pull request:
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #1456 
| ❓ Documentation  | https://github.com/snyk/driftctl-docs/pull/241

## Description

⚠️ Please review and test locally with attention

eliecharra committed:
feat: Add iam_group_policy_attachment

wbeuil commented on pkg/resource/aws/testdata/acc/aws_iam_group_policy/.driftignore:
```diff
@@ -1,2 +1,2 @@
 *
-!aws_iam_group_policy
+!aws_iam_group
```
I don't understand why changed the .driftignore of the `aws_iam_group_policy` test, can you explain?

wbeuil commented on pkg/resource/aws/testdata/acc/aws_iam_group_policy/terraform.tf:
```diff
@@ -29,4 +29,3 @@ resource "aws_iam_group_policy" "my_developer_policy" {
         ]
     })
 }
-
```
Useless noise here

wbeuil reviewed (changes requested):
Small comments and I think you may need to change back the terraform lock file of the group policy test and rerun the acc test of the group policy attachment one.

eliecharra commented:
@wbeuil I was rebased on an old version of the PR that added `aws_iam_group` #1477 that's why the diff was broken.

I removed the commit and rebased that one over main and thus that should be better now.

wbeuil reviewed (approved):
LGTM
